internal/controller/datadogagent/feature/admissioncontroller/feature.go

```diff
@@ -224,7 +231,62 @@ func (f *admissionControllerFeature) ManageDependencies(managers feature.Resourc
 	if err := managers.RBACManager().AddClusterPolicyRules(ns, rbacName, f.serviceAccountName, getRBACClusterPolicyRules(f.webhookName, f.cwsInstrumentationEnabled, f.cwsInstrumentationMode)); err != nil {
 		return err
 	}
-	return managers.RBACManager().AddPolicyRules(ns, rbacName, f.serviceAccountName, getRBACPolicyRules())
+	if err := managers.RBACManager().AddPolicyRules(ns, rbacName, f.serviceAccountName, getRBACPolicyRules()); err != nil {
+		return err
+	}
+
+	if f.networkPolicy != "" {
+		policyName, podSelector := objects.GetNetworkPolicyMetadata(f.owner, v2alpha1.ClusterAgentComponentName)
+		switch f.networkPolicy {
+		case v2alpha1.NetworkPolicyFlavorKubernetes:
+			ingressRules := []netv1.NetworkPolicyIngressRule{
+				{
+					Ports: []netv1.NetworkPolicyPort{
+						{
+							Port: &intstr.IntOrString{
+								Type:   intstr.Int,
+								IntVal: v2alpha1.DefaultAdmissionControllerTargetPort,
+							},
+						},
+					},
+				},
+			}
+			return managers.NetworkPolicyManager().AddKubernetesNetworkPolicy(
+				policyName,
+				f.owner.GetNamespace(),
+				podSelector,
+				nil,
+				ingressRules,
+				nil,
+			)
+		case v2alpha1.NetworkPolicyFlavorCilium:
+			policySpecs := []cilium.NetworkPolicySpec{
+				{
+					Description:      "Ingress from API server for admission controller",
+					EndpointSelector: podSelector,
+					Ingress: []cilium.IngressRule{
+						{
+							FromEntities: []cilium.Entity{
+								"kube-apiserver",
+							},
+							ToPorts: []cilium.PortRule{
+								{
+									Ports: []cilium.PortProtocol{
+										{
+											Port:     strconv.Itoa(v2alpha1.DefaultAdmissionControllerTargetPort),
```

Comment:
I was wondering if we have some available possibility to change the port, but it doesnt seems to be the case. Service is created the same way

Reply:
I see we have `DD_ADMISSION_CONTROLLER_PORT` in the agent but we don't have an option to set that in the operator. I can create a card for it